Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,610 @@
+	export tmp;
+}
+
+impliedval16: reg3_5 is reg3_5 & target3_5=6 & doublebyte=0 {
+	reg3_5 = reg3_5 - 1;
+	local tmp:2 = *:2 reg3_5;
+	export tmp;
+}
+
+impliedval16: reg3_5 is reg3_5 & (target3_5=4 | target3_5=5) & doublebyte=1 {
+	local tmp:2 = zext(*:1 reg3_5);
+	reg3_5 = reg3_5 + 1;
+	tmp = tmp | (zext(*:1 reg3_5) << 8);
+	reg3_5 = reg3_5 + 1;
+	export tmp;
+}
+
+impliedval16: reg3_5 is reg3_5 & target3_5=6 & doublebyte=1 {
+	reg3_5 = reg3_5 - 1;
+	local tmp:2 = zext(*:1 reg3_5);
+	reg3_5 = reg3_5 - 1;
+	tmp = tmp | (zext(*:1 reg3_5) << 8);
+	export tmp;
+}
+
+impliedval16: reg3_5 is reg3_5 & (target3_5=0 | target3_5=1 | target3_5=2 | target3_5=3 | target3_5=7) & doublebyte=1 {
+	local tmp:2 = zext(*:1 reg3_5);
+	tmp = tmp | (zext(*:1 reg3_5) << 8);
+	export tmp;
+}
+
+################################################################
+
+macro resultFlags(value) {
+	Z = value == 0;
+	S = value s< 0; 
+}
+
+macro addition(first, second) {
+	local tmpC = carry(first, second);
+	local tmpO = scarry(first, second);
+	first = first + second;
+	C = tmpC;
+	O = tmpO;
+	resultFlags(first);
+}
+
+macro comparison(first, second) {
+	local __val__ = first - second;
+	O = sborrow(first, second);
+	C = first < second; 
+	resultFlags(__val__);
+}
+
+macro subtraction(first, second) {
+	local __val__ = first - second;
+	O = sborrow(first, second);
+	C = first < second; 
+	resultFlags(__val__);
+	first = __val__;
+}
+
+################################################################
+
+define pcodeop TerminateCurrentInterrupt;
+define pcodeop SoftwareInterrupt;
+
+################################################################
+
+:ADD addr16, reg0_2 is opcode3_9=0x0058 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	addition(reg0_2, *:2 ptr);
+}
+
+:ADD@ impliedval16, reg0_2 is opcode6_9=0x000B & reg0_2 & impliedval16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	addition(reg0_2, impliedval16);
+}
+
+:ADCR reg0_2 is opcode3_9=0x0005 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local oldC = zext(C);
+	addition(reg0_2, oldC);
+}
+
+:ADDR reg3_5, reg0_2 is opcode6_9=0x0003 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	addition(reg0_2, reg3_5);
+}
+
+:AND addr16, reg0_2 is opcode3_9=0x0070 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	reg0_2 = reg0_2 & *:2 ptr;
+	resultFlags(reg0_2);
+}
+
+:AND@ impliedval16, reg0_2 is opcode6_9=0x000E & reg0_2 & impliedval16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 & impliedval16;
+	resultFlags(reg0_2);
+}
+
+:ANDR reg3_5, reg0_2 is opcode6_9=0x0006 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 & reg3_5;
+	resultFlags(reg0_2);
+}
+
+:B branchdest16 is (opcode6_9=0x0008 & branch_external=0 & branch_condition=0) ... & branchdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
```

Comment:
you might consider a table here for the condition for a single branch instruction, see x86 ( `cc:` with `^cc` )for an example
https://github.com/NationalSecurityAgency/ghidra/blob/bca65330a8b05c49655132ccdb93db9140568d76/Ghidra/Processors/x86/data/languages/ia.sinc#L987

Reply:
I tried this earlier on and I kept getting compilation errors by Sleigh, wonder what I was doing wrong back then.  Anyway, it is now as you suggested, thanks!